src/Application/Actions/Donations/Update.php

```diff
@@ -97,15 +97,14 @@ protected function action(Request $request, Response $response, array $args): Re
         while ($retryCount < self::MAX_UPDATE_RETRY_COUNT) {
             $this->entityManager->beginTransaction();
 
-            try {
-                $donation = $this->donationRepository->findAndLockOneBy(['uuid' => $args['donationId']]);
-
-                if (!$donation) {
-                    $this->entityManager->rollback();
+            $donation = $this->donationRepository->findAndLockOneBy(['uuid' => $args['donationId']]);
```

Comment:
moved this outside the following try/catch, because otherwise Psalm points out that $donation is possibly undefined when we try to use it in the catch block to the the uuid.